pkg/migrate/azure/dns.go

```diff
@@ -0,0 +1,1335 @@
+package azure
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/davecgh/go-spew/spew"
+
+	"github.com/sirupsen/logrus"
+
+	aznetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	azdns "github.com/Azure/azure-sdk-for-go/services/preview/dns/mgmt/2018-03-01-preview/dns"
+	azprivatedns "github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	azconfig "github.com/openshift/installer/pkg/asset/installconfig/azure"
+)
+
+type legacyDNSZoneSerialized struct {
+	zone       []byte
+	recordsets [][]byte
+}
+
+type legacyDNSZoneInfo struct {
+	zone       *azdns.Zone
+	recordsets []*azdns.RecordSet
+}
+
+type legacyDNSClient struct {
+	resourceGroup    string
+	public           bool
+	zonesClient      azdns.ZonesClient
+	recordsetsClient azdns.RecordSetsClient
+}
+
+func newLegacyDNSClient(session *azconfig.Session, resourceGroup string, public bool) *legacyDNSClient {
+	zonesClient := azdns.NewZonesClient(session.Credentials.SubscriptionID)
+	zonesClient.Authorizer = session.Authorizer
+
+	recordsetsClient := azdns.NewRecordSetsClient(session.Credentials.SubscriptionID)
+	recordsetsClient.Authorizer = session.Authorizer
+
+	return &legacyDNSClient{resourceGroup, public, zonesClient, recordsetsClient}
+}
+
+// safe pointer handling for int32
+func safeInt32(n *int32) int32 {
+	if n == nil {
+		return 0
+	}
+	return *n
+}
```

Comment:
i think you can use https://godoc.org/github.com/Azure/go-autorest/autorest/to

Reply:
Nice library, glad you pointed me to it. I've updated the code to use it!